stm32-modules/heater-shaker/include/heater-shaker/gcodes.hpp

```diff
@@ -116,48 +115,37 @@ struct GetTemperature {
     */
     using ParseResult = std::optional<GetTemperature>;
 
-    static auto write_response_into(char* buf, size_t available,
-                                    uint32_t current_temperature,
-                                    uint32_t setpoint_temperature) -> size_t {
+    template <typename InputIt, typename InLimit>
```

Comment:
Why are we changing these to templates?

Reply:
it's important for everything in the call chain from `run_once` to have similar templated parameters because that way you can both call it with `char*` in the firmware and with `string.begin()` in the test harness